### ISSUE TITLE
Error when Battery is low?!

### ISSUE THREAD

**Issue by h1nnak** [Ryochan7/DS4Windows#2297]
Hello,

i've never had this error before. My controller had less than 25% battery. The battery was probably empty and the program tried to send a notification and an error message was generated.. Unfortunately I can't deliver more ...

 - Controller: Nintendo Switch Pro Controller
 - Connection: **Bluetooth**
 - OS: W10 u2d
 - DS4Windows Version: 3.0.13

```
**Additional context**
Add any other context about the problem here.
2021-07-10 23:27:31.9957|INFO|DS4Windows version 3.0.13
2021-07-10 23:27:31.9957|INFO|DS4Windows exe file: DS4Windows.exe
2021-07-10 23:27:31.9957|INFO|DS4Windows Assembly Architecture: x64
2021-07-10 23:27:31.9957|INFO|OS Version: Microsoft Windows NT 10.0.19043.0
2021-07-10 23:27:31.9957|INFO|OS Product Name: Windows 10 Pro
2021-07-10 23:27:31.9957|INFO|OS Release ID: 2009
2021-07-10 23:27:31.9957|INFO|System Architecture: x64
2021-07-10 23:27:31.9957|INFO|Logger created
2021-07-10 23:27:33.1189|INFO|Running as User
2021-07-10 23:27:34.1460|INFO|Starting...
2021-07-10 23:27:34.1460|INFO|Using output KB+M handler: SendInput
2021-07-10 23:27:34.1460|INFO|Connection to ViGEmBus 1.17.333.0 established
2021-07-10 23:27:34.1460|INFO|Searching for controllers...
2021-07-10 23:27:34.1460|INFO|Using Shared Mode
2021-07-10 23:27:47.2329|INFO|Found Controller: E0:F6:B5:69:03:2A (BT) (Switch Pro)
2021-07-10 23:27:47.3369|INFO|Plugging in virtual X360 Controller
2021-07-10 23:27:47.3369|INFO|Associate X360 Controller in slot #1 for input Switch Pro controller #1
2021-07-10 23:27:47.3369|INFO|Output data to DS4 disabled. Lightbar and rumble events are not written to DS4 gamepad. If the gamepad is connected over BT then IdleDisconnect option is recommended to let DS4Windows to close the connection after long period of idling.
2021-07-10 23:27:47.7625|INFO|Controller 1 is using Profile "Pro Controller". Battery level is 25%
2021-07-10 23:50:27.5975|INFO|Disassociate X360 Controller from slot #1 for input Switch Pro controller #1
2021-07-10 23:50:27.5975|INFO|Unplugging virtual X360 Controller
2021-07-10 23:50:27.5975|INFO|Controller 1 was removed or lost connection. Charge the battery
2021-07-10 23:50:29.6044|INFO|Found Controller: E0:F6:B5:69:03:2A (BT) (Switch Pro)
2021-07-10 23:50:29.6212|INFO|Plugging in virtual X360 Controller
2021-07-10 23:50:29.6212|INFO|Associate X360 Controller in slot #1 for input Switch Pro controller #1
2021-07-10 23:50:29.6212|INFO|Output data to DS4 disabled. Lightbar and rumble events are not written to DS4 gamepad. If the gamepad is connected over BT then IdleDisconnect option is recommended to let DS4Windows to close the connection after long period of idling.
2021-07-10 23:50:29.9055|INFO|Controller 1 is using Profile "Pro Controller". Battery level is 0%
2021-07-10 23:50:32.6372|INFO|Disassociate X360 Controller from slot #1 for input Switch Pro controller #1
2021-07-10 23:50:32.6372|INFO|Unplugging virtual X360 Controller
2021-07-10 23:50:32.6372|INFO|Controller 1 was removed or lost connection. Charge the battery
2021-07-10 23:50:39.9450|INFO|Found Controller: E0:F6:B5:69:03:2A (BT) (Switch Pro)
2021-07-10 23:50:39.9605|INFO|Plugging in virtual X360 Controller
2021-07-10 23:50:39.9605|INFO|Associate X360 Controller in slot #1 for input Switch Pro controller #1
2021-07-10 23:50:39.9605|INFO|Output data to DS4 disabled. Lightbar and rumble events are not written to DS4 gamepad. If the gamepad is connected over BT then IdleDisconnect option is recommended to let DS4Windows to close the connection after long period of idling.

2021-07-10 23:50:41.6829|WARN|System.AggregateException: One or more errors occurred. (Ein an das System angeschlossenes Gerät funktioniert nicht.)
 ---> System.IO.IOException: Ein an das System angeschlossenes Gerät funktioniert nicht.
   at System.IO.FileStream.FileStreamCompletionSource.CompleteCallback(UInt64 packedResult)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pNativeOverlapped)
--- End of stack trace from previous location ---
   at System.IO.FileStream.FlushWriteBuffer(Boolean calledFromFinalizer)
   at System.IO.FileStream.FlushInternalBuffer()
   at System.IO.FileStream.Flush(Boolean flushToDisk)
   at System.IO.FileStream.Flush()
   at DS4Windows.InputDevices.SwitchProDevice.Subcommand(Byte subcommand, Byte[] tmpBuffer, UInt32 bufLen, Boolean checkResponse)
   at DS4Windows.InputDevices.SwitchProDevice.SetOperational()
   at DS4Windows.InputDevices.SwitchProDevice.StartUpdate()
   at DS4Windows.ControlService.HotPlug()
   at DS4WinWPF.DS4Forms.MainWindow.InnerHotplug2()
   at DS4WinWPF.DS4Forms.MainWindow.<WndProc>b__85_0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   --- End of inner exception stack trace ---
2021-07-10 23:51:00.1036|INFO|Request App Shutdown
2021-07-10 23:51:00.1036|INFO|Stopping Virtual Output Controllers
2021-07-10 23:51:00.1036|INFO|Closing connection to ViGEmBus
2021-07-10 23:51:00.1036|ERROR|Thread Crashed with message One or more errors occurred. (Das Gerät ist nicht angeschlossen.)
2021-07-10 23:51:00.1129|ERROR|System.AggregateException: One or more errors occurred. (Das Gerät ist nicht angeschlossen.)
 ---> System.IO.IOException: Das Gerät ist nicht angeschlossen.
   at System.IO.FileStream.WriteAsyncInternalCore(ReadOnlyMemory`1 source, CancellationToken cancellationToken)
   at System.IO.FileStream.FlushWriteAsync(CancellationToken cancellationToken)
   at System.IO.FileStream.FlushWriteBuffer(Boolean calledFromFinalizer)
   at System.IO.FileStream.FlushInternalBuffer()
   at System.IO.FileStream.Flush(Boolean flushToDisk)
   at System.IO.FileStream.Flush()
   at DS4Windows.InputDevices.SwitchProDevice.Subcommand(Byte subcommand, Byte[] tmpBuffer, UInt32 bufLen, Boolean checkResponse)
   at DS4Windows.InputDevices.SwitchProDevice.Detach()
   at DS4Windows.InputDevices.SwitchProDevice.DisconnectBT(Boolean callRemoval)
   at DS4Windows.ControlService.Stop(Boolean showlog, Boolean immediateUnplug)
   at DS4WinWPF.App.<>c.<CleanShutdown>b__46_0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at DS4WinWPF.App.CleanShutdown()
   at DS4WinWPF.App.Application_Exit(Object sender, ExitEventArgs e)
   at System.Windows.Application.OnExit(ExitEventArgs e)
   at System.Windows.Application.DoShutdown()
   at System.Windows.Application.ShutdownImpl()
   at System.Windows.Application.ShutdownCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by Ryochan7** [comment]
Both errors are related to trying to send a subcommand to a dead controller. Once during the Hotplug routine and the other error comes from DS4Windows trying to shutdown from the previous error and detach that controller. It should not affect normal usage for the most part but that type of error could be a bit annoying. Not sure how I would want to approach solving that issue right now.

**Comment by Ryochan7** [comment]
Hopefully this was fixed in version 3.0.14. I have not tested the scenario myself; need to play some more Ion Fury and drain my Switch Pro battery. I guess I will give it a try before I close this issue.
